cmd/s2i/main.go

```diff
@@ -356,28 +358,31 @@ func checkErr(err error) {
 
 func main() {
 	cfg := &api.Config{}
-	stiCmd := &cobra.Command{
+	s2iCmd := &cobra.Command{
 		Use: "s2i",
 		Long: "Source-to-image (S2I) is a tool for building repeatable docker images.\n\n" +
 			"A command line interface that injects and assembles source code into a docker image.\n" +
 			"Complete documentation is available at http://github.com/openshift/source-to-image",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if fileLogger, ok := glog.(*utilglog.FileLogger); ok {
+				fileLogger.SetLoggerLevel(*loglevelPtr)
+			}
+		},
```

Comment:
if i understand this correctly, you're setting the level at which messages go to the file logger (instead of glog) to the loglevelPtr value (which is the --loglevel value).  I don't think that's correct.  The original implementation split the messages at level 2.  (level 2 and below went to the file logger, level 3 and above went to glog).  that should still be the behavior.

however @smarterclayton and I did discuss that rather than splitting where some messages get reported via glog and some get reported via filelogger, it would be better if the wrapper logic said "if the loglevel is higher than 2, send everything to glog.  if the loglevel is <=2, send everything to filelogger", that way you don't have some messages getting glogged and some getting filelogged, in a single execution.  So if you want to implement that behavior, that would be good.  If not, we can do that in a follow up to this PR.